clickhouse-core/src/main/scala/com/clickhouse/spark/Utils.scala

```diff
@@ -186,4 +187,153 @@ object Utils extends Logging {
   def setTesting(): Unit = System.setProperty(IS_TESTING, "true")
 
   def isTesting: Boolean = System.getProperty(IS_TESTING) == "true"
+
+  object RuntimeDetector {
+
+    def detectRuntime(): Option[String] =
+      RuntimeDetector.detectViaStackTrace()
+        .orElse(RuntimeDetector.detectViaClassLoader())
+        .orElse(RuntimeDetector.detectViaThreadNames())
+
+    /**
+     * Examines the current stack trace and loaded classes for platform-specific signatures
+     */
+    def detectViaStackTrace(): Option[String] = {
+      val stackTrace = Thread.currentThread().getStackTrace
+      val stackClasses = stackTrace.map(_.getClassName.toLowerCase)
```

Comment:
I will convert the map to a set and just make exists and looks like it can applied to other detections